### PR TITLE
Heat equation on a sphere

### PR DESCRIPTION
**Pull request by bmxam**
Adapted from https://www.chebfun.org/examples/sphere/SphereHeatConduction.html

### CODE REVIEW

File: src/example/heat_equation_sphere/heat_equation_sphere.jl [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    θ = acos(z / r)
+    ϕ = sign(y) * acos(x / sqrt(x^2 + y^2))
+
+    # From "me" (adapted from wiki) -> physics
+    # θ = atan(z, sqrt(x^2 + y^2))
+    # ϕ = atan(y, x)
+
+    # Formulae from CoordinateTransformations
+    # θ = atan(y, x)
+    # ϕ = atan(z / sqrt(x^2 + y^2))
```
Comment (ghislainb) [inline src/example/heat_equation_sphere/heat_equation_sphere.jl:35]: cleaning ?

File: src/example/heat_equation_sphere/heat_equation_sphere.jl [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    )
+    projection_l2!(u, u0, mesh)
+
+    # Bilinear forms
+    m(u, v) = ∫(u ⋅ v)dΩ
+    a(u, v) = ∫(∇ₛ(u) ⋅ ∇ₛ(v))dΩ
+
+    # Assemble
+    M = assemble_bilinear(m, U, V)
+    K = assemble_bilinear(a, U, V)
```
Comment (ghislainb) [inline src/example/heat_equation_sphere/heat_equation_sphere.jl:180]: add `factorize` step ?

Comment (bmxam) [reply src/example/heat_equation_sphere/heat_equation_sphere.jl:180]: no need since it's a steady resolution, so the inversion will occur only once

File: src/example/heat_equation_sphere/heat_equation_sphere.jl [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    out_dir = joinpath(@__DIR__, "../../../myout/heat_eqn_sphere")
+    mkpath(out_dir)
+
+    # Mesh
+    mesh_path = joinpath(out_dir, "mesh.msh")
+    Bcube.gen_sphere_mesh(mesh_path; radius = 1.0, lc = lc)
+    mesh = read_msh(mesh_path)
+    rng = MersenneTwister(0)
+    R = rotMat(rand(rng, 3)...)
+    transform!(mesh, x -> R * x) # rotate to avoid being "aligned" with an axis
```
Comment (lokmanb) [inline src/example/heat_equation_sphere/heat_equation_sphere.jl:140]: Why is this required?

File: src/example/heat_equation_sphere/heat_equation_sphere.jl (hunk 2 above)
Comment (lokmanb) [reply src/example/heat_equation_sphere/heat_equation_sphere.jl:171]: What do you mean by steady resolution?

**Review by lokmanb — APPROVED** [review]
Really cool example!

File: src/example/heat_equation_sphere/heat_equation_sphere.jl (hunk 2 above)
Comment (bmxam) [reply src/example/heat_equation_sphere/heat_equation_sphere.jl:171]: Well, you're right it's not steady. I mean there is only one time step, i.e one linear system inversion. So it's not necessary to factorize.

File: src/example/heat_equation_sphere/heat_equation_sphere.jl (hunk 3 above)
Comment (bmxam) [reply src/example/heat_equation_sphere/heat_equation_sphere.jl:140]: Good question. The coordinate mapping is "discontinuous", the angles jumping from 0 to 2pi. If the mesh is aligned with an axis, the dofs lying on the axis are not correctly computed ~ you see the discontinuity.
